rust/lance/src/dataset/optimize/remapping.rs

```diff
@@ -257,6 +260,9 @@ async fn remap_index(dataset: &mut Dataset, index_id: &Uuid) -> Result<()> {
                 dataset_version: dataset.manifest.version,
                 fragment_bitmap: bitmap_after_remap,
                 index_details: curr_index_meta.index_details.clone(),
+                index_version: infer_index_type(&curr_index_meta)
+                    .map(|index_type| index_type.version())
+                    .unwrap_or_default(),
```

Comment:
Do we need to add `FragmentReuse` to the version enum switch like Jack has here? https://github.com/lancedb/lance/pull/3911/files